paymentapi/src/main/java/fi/hel/verkkokauppa/payment/service/PaymentMethodService.java

```diff
@@ -0,0 +1,233 @@
+			)).toArray(PaymentMethodDto[]::new);
+
+		} catch (RuntimeException e) {
+			log.warn("getting offline payment methods failed, currency: " + currency, e);
+			return new PaymentMethodDto[0];
+		}
+	}
+
+	public boolean isDefaultCurrency(String currency) {
+		return Objects.equals(currency, CurrencyUtil.DEFAULT_CURRENCY);
+	}
+
+	public PaymentMethodDto[] filterPaymentMethodList(GetPaymentMethodListRequest request, PaymentMethodDto[] methods) {
+		Set<PaymentMethodDto> filteredMethodsList = new HashSet<>();
+
+		// If namespace has multiple filters, available methods for all of them will be returned
+		List<PaymentMethodFilter> filtersForNamespace = getFiltersEnabledForNamespace(request.getNamespace());
+
+		for (PaymentMethodFilter paymentMethodFilter : filtersForNamespace) {
+			String filterKey = getFilterKeys(paymentMethodFilter, request);
+
+			HashMap<String, List<String>> filterValues = getFilterValuesMap(paymentMethodFilter);
+
+			for (Map.Entry<String, List<String>> entry : filterValues.entrySet()) {
+				String key = entry.getKey();
+
+				if (filterKey.equalsIgnoreCase(key)) {
+					List<String> paymentMethodGroups = entry.getValue();
+
+					for (String value : paymentMethodGroups) {
+						filteredMethodsList.addAll(Arrays.stream(methods)
+								.filter(method -> method.getGroup().equalsIgnoreCase(value))
+								.collect(Collectors.toList()));
+					}
+				}
+			}
+		}
+
+		if (!filteredMethodsList.isEmpty()) {
+			methods = filteredMethodsList.toArray(new PaymentMethodDto[0]);
+		}
+
+		return methods;
+	}
+
+	private List<PaymentMethodFilter> getFiltersEnabledForNamespace(String namespace) {
+		List<PaymentMethodFilter> filtersForNamespace = new ArrayList<>();
+
+		for (PaymentMethodFilter methodFilter : PaymentMethodFilter.getAll()) {
+			String orderTypeFilterEnabledList = env.getRequiredProperty("enabled_namespaces.payment_method_filter." + methodFilter);
+
+			if (orderTypeFilterEnabledList.contains(namespace)) {
+				filtersForNamespace.add(methodFilter);
+			}
+		}
+
+		return filtersForNamespace;
+	}
+
+	public String getFilterKeys(PaymentMethodFilter filter, GetPaymentMethodListRequest request) {
+		OrderDto orderDto = request.getOrderDto();
+
+		if (filter.equals(PaymentMethodFilter.ORDER_TYPE)) {
+			return orderDto.getType();
+		}
+
+		throw new CommonApiException(
+				HttpStatus.INTERNAL_SERVER_ERROR,
+				new Error("unknown-payment-method-filter-type", "unknown payment method filter type")
+		);
+	}
+
+	// namespace can be used as a parameter if different namespaces need to return different values for filter type
+	public HashMap<String, List<String>> getFilterValuesMap(PaymentMethodFilter filter) {
+		HashMap<String, List<String>> values = new HashMap<>();
+
+		if (filter.equals(PaymentMethodFilter.ORDER_TYPE)) {
+			values.put(OrderType.SUBSCRIPTION, Collections.singletonList(PaymentType.CREDIT_CARDS));
+		}
+
+		return values;
+	}
+
+	public List<PaymentMethodDto> getAllPaymentMethods() {
```

Comment:
Below are the rest of my changes.

Reply:
fi.hel.verkkokauppa.payment.model must come from import statement, easier to read and it keeps same convention between project coding style